drivers/gpio/gpio_mmio32.c

```diff
@@ -35,27 +35,34 @@ static int gpio_mmio32_config(struct device *dev, int access_op,
 	struct gpio_mmio32_context *context = dev->driver_data;
 	const struct gpio_mmio32_config *config = context->config;
 
-	if (flags & GPIO_INT) {
-		return -ENOTSUP;
-	}
-
-	if (access_op != GPIO_ACCESS_BY_PIN) {
-		return -ENOTSUP;
-	}
-
 	if ((config->mask & (1 << pin)) == 0) {
 		return -EINVAL; /* Pin not in our validity mask */
 	}
 
-	if (flags & ~(GPIO_DIR_MASK | GPIO_POL_MASK)) {
+	if (flags & ~(GPIO_INPUT | GPIO_OUTPUT |
+		      GPIO_OUTPUT_INIT_LOW | GPIO_OUTPUT_INIT_HIGH |
+		      GPIO_ACTIVE_LOW)) {
 		/* We ignore direction and fake polarity, rest is unsupported */
 		return -ENOTSUP;
 	}
 
-	if ((flags & GPIO_POL_MASK) == GPIO_POL_INV) {
-		context->invert |= (1 << pin);
-	} else {
-		context->invert &= ~(1 << pin);
+	if ((flags & GPIO_OUTPUT) != 0) {
+		unsigned int key;
+		volatile u32_t *reg = config->reg;
+		u32_t mask = config->mask;
+
+		mask &= 1 << pin;
+		if (!mask) {
+			return -EINVAL; /* Pin not in our validity mask */
+		}
```

Comment:
This check seems to be independent of the direction, should probably be moved out of the if statement. To be Misra-C compliant we could write it as `if ((mask & (1 << pin)) == 0) {`.

Reply:
It already is, so removing the duplicated check ;)